iphone/Classes/KrollContext.m

```diff
@@ -1560,6 +1560,82 @@ - (void)dealloc
 
 #if defined(USE_JSCORE_FRAMEWORK) && !defined(TI_USE_KROLL_THREAD)
 
+@class KrollTimerManager;
+
+/**
+ * Object acting as the target that receives a message when a timer fires.
+ */
+@interface KrollTimerTarget : NSObject
```

Comment:
So, looks like you and I are converging together to very similar solutions here. You're right, we don't need to use JSManagedValues, I was wrong about that. That actually ended up causing deadlocks which is why my PRs keep timing out on tests.

Feels to me like a little overkill to hold all of this in a new type. We're already holding the timer id/map in the timer manager. The callback (and also arguments) could be passed in as the "user info".

Single-shot timers will get invalidated automatically after first being fired, so I'm not sure we really need to retain the repeat value in user info. I suppose it means we then are better about cleaning up our map after firing? My "solution" here was to actually use a strong key to weak object map, because the run loop already holds a strong reference to the timer and invalidates/cleans it up once it's done. So in my implementation the weak pointer lets it do that, and really only the NSNumber key is left "alive".

I did simplify and ignore the rare use case of additional arguments, but I suppose I could just pass in the whole argument array as the `userInfo` and pull the callback args apart when the timer fired.